src/components/AnimateSharedLayout/SharedLayoutTree.tsx

```diff
@@ -0,0 +1,108 @@
+import * as React from "react"
+import {
+    SharedLayoutContext,
+    SharedLayoutSyncMethods,
+} from "./SharedLayoutContext"
+import { HTMLVisualElement } from "render/dom/HTMLVisualElement"
+import { Presence } from "./types"
+
+export interface SharedLayoutTreeProps {
+    /**
+     * @public
+     */
+    children: React.ReactNode
+
+    /**
+     * Are the children of the tree valid targets when computing lead and follow.
+     */
+    isPresent: boolean | undefined
+    /**
+     * Are the children of the tree valid targets when computing lead and follow.
+     */
+    isValidTree: boolean
+
+    /**
+     * What priority should children of this tree take when determining the lead and follow.
+     */
+    layoutOrder: number
+
+    isLead: boolean
```

Comment:
This is KINDA just here to trigger an update on the tree we're animating too in the scenario where the layoutOrder doesn't update.